### PR TITLE
Refactor Google sign-in settings page to use Form component

### LINKED ISSUE #11659: Errors on Google Auth settings page don't look great
Screenshot: image

The error in the screenshot was triggered by changing and saving the email domain value multiple times.

@tlrobinson mentioned that this form should be ported to the new form code.

### LINKED ISSUE #15974: Google sign-in client ID setting regularly fails to save
**Describe the bug**
When trying to set the client ID field for the Google sign-in settings, it fails about half the time. Failures manifest as a page refresh after clicking "save changes", and the old value still being present in the client ID field. This isn't deterministic, since sometimes I'll retry saving a value 2-3 times before it succeeds. It also doesn't have any bearing on whether the value is a valid client ID.

Browser console also logs `error updating setting` every time this form is submitted, regardless of whether saving succeeded or not.

**To Reproduce**
1. Go to the Admin Panel
2. Click on Authentication
3. Under "Sign in with Google" click "Configure"
4. Put any value into the first text box and click "Save Changes"
5. If the value was saved successfully after the page refreshes, retry step 4 with different values

**Expected behavior**
Value should save successfully every time the form is submitted.

**Information about your Metabase Installation:**
Firefox 88.0.1 on MacOS Big Sur, using an H2 database

### LINKED ISSUE #16203: Different case in "Save changes" button between Google vs LDAP settings
**Describe the bug**
Google UI uses "Save Changes" with the capital "C", while LDAP uses "Save changes" (lower case "c")

**Expected behavior**
We should stick to the one case and apply it consistently.

**Additional context**
Accidentally discovered this while writing e2e tests around LDAP and saw them failing around a helper function:
```javascript
function saveSettings() {
  cy.button("Save Changes").click();
  cy.findByText("Saved");
}
```

### PR DESCRIPTION
**Pull request by noahmoss**
Deletes `SettingsSingleSignOnForm` (which was only used for Google sign-in settings) and creates `SettingsGoogleForm` that uses the more standard `Form` components. I used the existing `SettingsSAMLForm` as reference for the new component.

This resolves #15974, #11659 and #16203. It also makes it possible to use the plugin system to alter the description of a single field between OSS and EE, which I'll be using for #15820.

The UI looks a bit different now, since it's more consistent with other setting forms:
* Different styling on input fields and submit button
* Bold titles on fields
* No `@` symbol before domain input. Couldn't figure out how to do this but I can add it back if any FE engineers know how.
* Removed the line about using the full client ID and instead added illustrating placeholder text. This was discretionary on my part since it seemed cleaner, but I'd love feedback from design on this.
* Added placeholder example domain

Before             |  After
:-------------------------:|:-------------------------:
  |

### CODE REVIEW

File: frontend/src/metabase/admin/settings/components/SettingsGoogleForm.jsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+              target="_blank"
+            >
+              {t`here`}
+            </ExternalLink>
+          )}.`}
+        </p>
+        <FormField
+          {...settingField("google-auth-client-id")}
+          title={t`Client ID`}
+          description=""
```
Comment (kulyk) [inline frontend/src/metabase/admin/settings/components/SettingsGoogleForm.jsx:63]: nip: can we just remove the `description` prop? Or does `<FormField />` have some non-empty default value? 🤔

**Review by kulyk — APPROVED** [review]
Looks good!

File: frontend/src/metabase/admin/settings/components/SettingsGoogleForm.jsx (hunk 1 above)
Comment (noahmoss) [reply frontend/src/metabase/admin/settings/components/SettingsGoogleForm.jsx:63]: `settingField` includes a description prop that defaults to the description returned by the backend, so we need to override it with an empty string.

Comment (kulyk) [reply frontend/src/metabase/admin/settings/components/SettingsGoogleForm.jsx:63]: Oh, sorry I didn't notice the `settingField` spread above. Thank you!

File: frontend/src/metabase/admin/settings/components/SettingsGoogleForm.jsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
+/* eslint-disable react/prop-types */
```
Comment (nemanjaglumac) [inline frontend/src/metabase/admin/settings/components/SettingsGoogleForm.jsx:1]: Let's please not introduce new prop type errors (even if we ignore them). This should be handled before the merge.

Comment (noahmoss) [reply frontend/src/metabase/admin/settings/components/SettingsGoogleForm.jsx:1]: Ah, sorry, I copied this bit from `SettingsSAMLForm`. I'll change it.

**Comment by mazameli** [comment]
UI changes seem cool to me, but did the form previously have any validation? I can type gibberish into the cilent ID and it saves successfully. If it didn't previously have validation, I can open that as a new issue.

form-validation

**Comment by noahmoss** [comment]
@mazameli Yeah, that's not new. I'm planning on adding some basic validation soon in a separate PR.

**Comment by nemanjaglumac** [comment]
Repros for some of the fixed issues are still in the `release-x.39.x`. Mental note to myself to unskip them once they are merged into `master`.

**Pull request merged**